crdb-seed/src/main.rs

```diff
@@ -0,0 +1,71 @@
+use camino::Utf8PathBuf;
+use dropshot::{test_util::LogContext, ConfigLogging, ConfigLoggingLevel};
+use omicron_test_utils::dev;
+use slog::Logger;
+use std::io::Write;
+
+// Creates a string identifier for the current DB schema and version.
+//
+// The goal here is to allow to create different "seed" directories
+// for each revision of the DB.
+fn digest_unique_to_schema() -> String {
+    let schema = include_str!("../../schema/crdb/dbinit.sql");
+    let crdb_version = include_str!("../../tools/cockroachdb_version");
+    let mut ctx = ring::digest::Context::new(&ring::digest::SHA256);
+    ctx.update(&schema.as_bytes());
+    ctx.update(&crdb_version.as_bytes());
+    let digest = ctx.finish();
+    hex::encode(digest.as_ref())
+}
+
+async fn ensure_seed_directory_exists(log: &Logger) -> Utf8PathBuf {
+    let base_seed_dir = Utf8PathBuf::from_path_buf(std::env::temp_dir())
+        .expect("Not a UTF-8 path")
+        .join("crdb-base");
+    std::fs::create_dir_all(&base_seed_dir).unwrap();
+    let desired_seed_dir = base_seed_dir.join(digest_unique_to_schema());
+
+    // If the directory didn't exist when we started, try to create it.
+    //
+    // Note that this may be executed concurrently by many tests, so
+    // we should consider it possible for another caller to create this
+    // seed directory before we finish setting it up ourselves.
```

Comment:
Is this still true, or will nextest prevent the concurrent execution of this function?

Reply:
Nextest will execute it once, but it's possible that the user spins up two instances of nextest at the same time. I'll update the comment.